src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -422,33 +422,34 @@ public static void main(String args[]){
        
       else if(args[0].equals("runByPriority")){
         int priority = Integer.parseInt(args[1]);
-        if(priority<0||priority>2) {
-          throw new IllegalArgumentException("Illegal priority level!");
+        if(priority < 0 || priority > 2) {
+          throw new IllegalArgumentException("Invalid priority level!");
         }
-        myTestHarness.runByPriority(priority);
+        testHarness.runByPriority(priority);
       }
        
       else if(args[0].equals("runAll")){
-        if(args.length>1){
+        if(args.length > 1){
           throw new IllegalArgumentException("More arguments provided than placeholders specified!");          
         }
         else{
-          myTestHarness.runAll();
+          testHarness.runAll();
         }
       }
        
       else if(args[0].equals("runByName")){
+        File testFile = new File(args[1]); 
         if(!args[1].startsWith("src/")){
           throw new IllegalArgumentException("Test file path doesn't start from src/");
         }
         else if(!args[1].endsWith("Test.java"){
           throw new IllegalArgumentException("Test file path doesn't end with Test.java!");
         }
-        else if(testFile.exists()&&testFile.isFile()){
+        else if(testFile.exists() && testFile.isFile()){
           throw new IllegalArgumentException("Test file doesn't exist!");
```

Comment:
Throwing exception when testFile exists and is a file

Reply:
it will be isDirectory good catch

Reply:
or !isFile()